central/auth/datastore/datastore_impl.go

```diff
@@ -147,6 +151,28 @@ func (d *datastoreImpl) InitializeTokenExchangers() error {
 			continue
 		}
 	}
+
+	if features.PolicyAsCode.Enabled() {
+		kubeSAIssuer, err := d.issuerFetcher.GetServiceAccountIssuer()
+		if err != nil {
+			return errors.Join(tokenExchangerErrors, err)
+		}
+
+		// Unconditionally add K8s service account exchanger
+		// This is required for config-controller auth
+		err = d.set.UpsertTokenExchanger(ctx, &storage.AuthMachineToMachineConfig{
```

Comment:
Refactoring suggestion: add this `storage.AuthMachineToMachineConfig` to the `configs` at about line 146 under the feature flag condition.